### ISSUE TITLE
Does RN StyleX limit performance?

### ISSUE THREAD

**Issue by smlarkin** [retyui/react-native-stylex#9]
When creating a `StyleSheet`, outside of a component and then referencing its styles in the component, RN creates a pointer to those styles so they do not have to be recalculated on every update.

By using the useStyles hook, it seems like you might need to recalculate all styles each time the props are updated like inline styles ...

https://stackoverflow.com/questions/39336266/react-native-inline-styles-and-performance

...Is this the case OR does the hook preserve the original calculation when the component first renders?

**Comment by ElForastero** [comment]
Yes, due to usage of `StyleSheet.create()` inside Component's body, performance will be slower.
This is why I created [@pulsar/core](https://github.com/ElForastero/pulsar-core)

**Comment by retyui** [comment]
> Is this the case OR does the hook preserve the original calculation when the component first renders?

Of Course, the module memoize styles and **recalculated them only when dependencies will be updated**

Also was added a **mechanism to detect what exactly was changed** (theme \ appearance \ window dimensions) to avoid redundant communication `JS<==>Native`

```js
import { makeUseStyles } from 'react-native-stylex';
import { maxWidth } from 'react-native-stylex/media-query';

// update styles when `Dimensions.get('window')` will be changed
const useStyles = makeUseStyles(() => ({
  point: maxWidth(320, { height: 4, width: 4 }),
}));

// update styles when `theme` will be changed
const useStyles = makeUseStyles(({ palette, utils }) => ({
  root: {
    color: utils.fade(palette.text.textColor, 0.5),
  },
}));
```

**Comment by smlarkin** [comment]
@retyui I think what @ElForastero is pointing out is the initial cost of calling the `useStyles` hook inside the component.  I believe with a vanilla `StyleSheet`, any (non-dynamic) styles are calculated at build time and available for consumption on the component.  But when a hook is used to generate the styles inside of the component body, then they are recalculated every time the component mounts (not updates) ... even if you memoized the initial styles.

I point this out because I wanted to use Tailwind RN, but it is super inefficient since every mount and update it has to recalculate all component styles.

**Comment by retyui** [comment]
Thank you for reporting this issue will be fixed in the new version [#10](https://github.com/retyui/react-native-stylex/pull/10)
